core/store/src/trie/iterator.rs

```diff
@@ -212,7 +220,11 @@ impl<'a> TrieIterator<'a> {
         let mut nodes_list = Vec::new();
         // Actually (self.key_nibbles[..] == path_begin) always because path_begin always ends in a node
         if &self.key_nibbles[..] >= path_begin {
-            nodes_list.push(last_hash);
+            let is_contract_code_node = match self.current_iter_step() {
+                Some(IterStep::Value(_)) => is_contract_code_key(&self.key()),
+                _ => false,
+            };
+            nodes_list.push((last_hash, is_contract_code_node));
```

Comment:
Don't think it is great to have this function care about some specific trie key. It may make sense to pass some function here as an argument.